### PR TITLE
More verbose error messages

### PR DESCRIPTION
**Pull request by FedericoV**
Makes the error messages a bit more verbose in a few places, with information about types and values.

### CODE REVIEW

File: diffrax/integrate.py [hunk 1]
```diff
@@ -599,7 +599,8 @@ def diffeqsolve(
 
     # Error checking
     if dt0 is not None:
-        error_if(lambda: (t1 - t0) * dt0 <= 0, "Must have (t1 - t0) * dt0 > 0")
+        msg = f"Must have (t1 ({t1, type(t1)}) - t0  ({t0, type(t0)})) * dt0  ({dt0, type(dt0)})> 0"
```
Comment (patrick-kidger) [inline diffrax/integrate.py:602]: So I find this a little hard to read. Can the `f"{}"` values be collected at the end of the error message?
When printing out a type, can we use plain English to denote what's going on, e.g. `Got value {t1} of type {type(t1)}.`

File: diffrax/misc/errors.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def raises(_arg):
         raises((pred, index))
     else:
         msg = (
-            "`pred` must either be a `bool`, a NumPy array, a JAX array, or a "
-            "zero-argument callable that returns a `bool` or JAX array."
+            f"`pred` {pred} must either be a `bool`, a JAX array, or a zero-argument callable "
+            f"that returns a `bool` or JAX array, instead of {type(pred)}."
```
Comment (patrick-kidger) [inline diffrax/misc/errors.py:61]: I'd don't hold to this as a hard-and-fast rule, but for something like this where an error here can almost only ever occur due to something totally crazy: I'd recommend putting the `f"{}"` values at the end of the docstring. That way if they're something with a huge `__str__` they won't blat the rest of the error message that comes after them.

**Review by patrick-kidger — COMMENTED** [review]
I apologise for doing [this](https://twitter.com/iamdevloper/status/397664295875805184) ;)

File: diffrax/misc/errors.py (hunk 2 above)
Comment (FedericoV) [reply diffrax/misc/errors.py]: All good :)

Let me know if this looks better.

**Comment by patrick-kidger** [comment]
LGTM! I'll merge once the tests pass and we'll fold this into the 0.0.3 release.

**Comment by patrick-kidger** [comment]
...*if* the tests pass, haha.

It looks you haven't set up the pre-commit hooks. See [CONTRIBUTING.md](https://github.com/patrick-kidger/diffrax/blob/main/CONTRIBUTING.md).

**Pull request merged**